### PR TITLE
Remove old version metrics collector

### PR DESCRIPTION
**Pull request by hougangliu**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/761)

### CODE REVIEW

File: pkg/util/v1alpha3/katibclient/katib_client.go [hunk 1]
```diff
@@ -41,7 +41,6 @@ type Client interface {
 	GetTrialList(name string, namespace ...string) (*trialsv1alpha3.TrialList, error)
 	GetTrialTemplates(namespace ...string) (map[string]string, error)
 	UpdateTrialTemplates(newTrialTemplates map[string]string, namespace ...string) error
-	GetMetricsCollectorTemplates(namespace ...string) (map[string]string, error)
 	UpdateMetricsCollectorTemplates(newMCTemplates map[string]string, namespace ...string) error
```
Comment (johnugeorge) [inline pkg/util/v1alpha3/katibclient/katib_client.go]: Is UpdateMetricsCollectorTemplates  needed?

File: examples/v1alpha3/random-example.yaml [hunk 2]
```diff
@@ -26,13 +26,15 @@ spec:
           spec:
             template:
               spec:
+                serviceAccountName: metrics-collector
```
Comment (gaocegege) [inline examples/v1alpha3/random-example.yaml]: I am not sure why do we need to set service account for the trial

File: pkg/util/v1alpha3/katibclient/katib_client.go (hunk 1 above)
Comment (hougangliu) [reply pkg/util/v1alpha3/katibclient/katib_client.go]: delete it, too
Thanks

File: examples/v1alpha3/random-example.yaml (hunk 2 above)
Comment (hougangliu) [reply examples/v1alpha3/random-example.yaml]: for now, metrics-collector get logs from k8s api, it need some role. I will delete it after implement file reading

Comment (gaocegege) [reply examples/v1alpha3/random-example.yaml]: SGTM

**Pull request merged**